src/hipscat/io/parquet_metadata.py

```diff
@@ -0,0 +1,109 @@
+"""Utility functions for handling parquet metadata files"""
+import tempfile
+from typing import List
+
+import pyarrow as pa
+import pyarrow.dataset as pds
+import pyarrow.parquet as pq
+
+from hipscat.io import file_io, paths
+from hipscat.io.file_io.file_pointer import get_fs, strip_leading_slash_for_pyarrow
+
+
+def row_group_stat_single_value(row_group, stat_key):
+    """Convenience method to find the min and max inside a statistics dictionary,
+    and raise an error if they're unequal."""
+    if stat_key not in row_group.statistics:
+        raise ValueError(f"row group doesn't have expected key {stat_key}")
+    stat_dict = row_group.statistics[stat_key]
+    min_val = stat_dict["min"]
+    max_val = stat_dict["max"]
+    if min_val != max_val:
+        raise ValueError(f"stat min != max ({min_val} != {max_val})")
+    return min_val
+
+
+def write_parquet_metadata(catalog_path, storage_options: dict = None, output_path: str = None):
+    """Generate parquet metadata, using the already-partitioned parquet files
+    for this catalog
+
+    Args:
+        catalog_path (str): base path for the catalog
+        storage_options: dictionary that contains abstract filesystem credentials
+        output_path (str): base path for writing out metadata files
+            defaults to `catalog_path` if unspecified
+    """
+    ignore_prefixes = [
+        "intermediate",
+        "_common_metadata",
+        "_metadata",
+    ]
+
+    dataset = file_io.read_parquet_dataset(
+        catalog_path,
+        storage_options=storage_options,
+        ignore_prefixes=ignore_prefixes,
+        exclude_invalid_files=True,
+    )
+    metadata_collector = []
+
+    for hips_file in dataset.files:
+        hips_file_pointer = file_io.get_file_pointer_from_path(hips_file, include_protocol=catalog_path)
+        single_metadata = file_io.read_parquet_metadata(hips_file_pointer, storage_options=storage_options)
+        relative_path = hips_file[len(catalog_path) :]
+        single_metadata.set_file_path(relative_path)
+        metadata_collector.append(single_metadata)
```

Comment:
This is more of a question because I am a bit unfamiliar with pyarrow! Here we're reading the catalog parquet files metadata and setting the metadata object file path to something like `Norder=X,Dir=Y,Npix=Z.parquet`? If you could please add a brief comment here explaining why this is needed I think it'd be useful. Thanks!

Reply:
That's exactly what this code chunk is doing. I'm not sure why, but the parquet dataset API does NOT fill in the file path per row group metadata, and leaves it up to the user to do so. 

They mention it off-handedly in this page about writing _metadata files:
https://arrow.apache.org/docs/python/parquet.html#writing-metadata-and-common-metadata-files